### PR TITLE
Make sure MSI build works when not preview

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Make sure MSI build works when not preview

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by adityapatwardhan — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]

File: assets/Product.wxs [hunk 1]
```diff
@@ -165,7 +165,7 @@
             <!-- add ourselves to %PATH% so pwsh.exe can be started from Windows PowerShell or cmd.exe -->
             <Component Id="SetPath" Guid="{9dbb7763-7baf-48e7-b025-3bdedcb0632f}" KeyPath="yes">
               <Condition>ADD_PATH=1</Condition>
-              <Environment Id="PATH" Action="set" Name="PATH" Part="last" Permanent="no" System="yes" Value="[$(var.ProductDirectoryName)]$(var.PwshPath)"/>
+              <Environment Id="PATH" Action="set" Name="PATH" Part="last" Permanent="no" System="yes" Value="$(var.PwshPath)"/>
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply assets/Product.wxs]: The variable is set to empty in the non-preview case and the build fails

**Comment by TravisEz13** [comment]
@bergmeister  The variable is set to empty (meaning to delete the variable) in the non-preview case and the build fails

**Comment by TravisEz13** [comment]
used admin to squash when windows CI was finished as this did not affect other platforms

**Comment by TravisEz13** [comment]
@bergmeister feel free to continue the discussion.  This was blocking the 6.1 builds so I merged so it's easier to take for the build.

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -2521,14 +2521,14 @@ function New-MSIPackage
...
-        [Environment]::SetEnvironmentVariable("PwshPath", '', "Process")
+        [Environment]::SetEnvironmentVariable("PwshPath", "[$productDirectoryName]", "Process")
         [Environment]::SetEnvironmentVariable("UpgradeCodeX64", '31ab5147-9a97-4452-8443-d9709f0516e1', "Process")
         [Environment]::SetEnvironmentVariable("UpgradeCodeX86", '1d00683b-0f84-4db8-a64f-2f98ad42fe06', "Process")
         [Environment]::SetEnvironmentVariable("IconPath", 'assets\Powershell_black.ico', "Process")
     }
     else
     {
-        [Environment]::SetEnvironmentVariable("PwshPath", 'preview', "Process")
+        [Environment]::SetEnvironmentVariable("PwshPath", "[$productDirectoryName]preview", "Process")
```
Comment (bergmeister) [inline tools/packaging/packaging.psm1]: @TravisEz13 Non-blocking comment: Would it not be cleaner if a new variable was declared to not convolute the script with WiX syntax?

Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: This variable was already there and only used for this purpose.  I don't think I'm getting the meaning of your statement.

Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: I do think there is probably a cleaner way of doing this.  Like refering to the preview directory by name, but the goal here was to make a quick change that would not require a lot of re-testing for the 6.1.0 release.

Comment (bergmeister) [reply tools/packaging/packaging.psm1]: `[$productDirectoryName]preview` is WiX syntax that we inject into the WiX file. The build script should be generic and know as little about WiX as possible to reduce coupling if installer technology changed for example or WiX 4 brings breaking changes in syntax. How about this:
```powershell
[Environment]::SetEnvironmentVariable("PwshPath", (Join-Path $productDirectoryName 'preview'), "Process")
```

Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: I'm fine with that change for 6.2.

**Pull request merged**